src/metabase/search/impl.clj

```diff
@@ -247,14 +244,263 @@
+         ;; TODO: update build-filters to only query from the search table, and not use joins
+         (search.filter/build-filters model search-ctx)
+         ; TODO: implement bookmarks
+         #_(sql.helpers/left-join [:card_bookmark :bookmark]
+                                  [:and
+                                   [:= :bookmark.card_id :card.id]
+                                   [:= :bookmark.user_id (:current-user-id search-ctx)]])
+         ; TODO: implement collection filtering
+         #_(add-collection-join-and-where-clauses "card" search-ctx)
+         ; TODO: implement table-db-id fitler
+         #_(add-card-db-id-clause (:table-db-id search-ctx))))))
+
+(defmethod searchable-data-query-for-model "table"
+  [model]
+  {:query (let [table-column #(search.config/column-with-alias model %)]
+            (-> (-> {:select [[[:inline "table"] :search_model]
+                              [:table.id :id]
+                              [:table.name :name]
+                              [:table.created_at :created_at]
+                              [:table.display_name :display_name]
+                              [:table.description :description]
+                              [:table.updated_at :updated_at]
+                              [:table.initial_sync_status :initial_sync_status]
+                              [:table.id :table_id]
+                              [:table.db_id :database_id]
+                              [:table.schema :table_schema]
+                              [:table.name :table_name]
+                              [:table.description :table_description]
+                              [:metabase_database.name :database_name]]
+                     :from   (from-clause-for-model model)}
+              ;; TODO
+                    #_(search.filter/build-filters model context))
+                (sql.helpers/where [:and
+                                    [:not [:= :table.db_id [:inline audit/audit-db-id]]]
+                                    [:= :table.active true]
+                                    [:= :table.visibility_type nil]])
+                (sql.helpers/left-join :metabase_database [:= (table-column :db_id) :metabase_database.id])))
+   :columns [:search_model
+             :id
+             :name
+             :created_at
+             :display_name
+             :description
+             :updated_at
+             :initial_sync_status
+             :table_id
+             :database_id
+             :table_schema
+             :table_name
+             :table_description
+             :database_name]})
+
+(defn searchable-data-query-for-instance [model id]
+  (let [query (:query (searchable-data-query-for-model model))]
+    (sql.helpers/where query [:= :id id])))
+
+(defn search-all-models-query [search-ctx]
+  {:select [:*]
+   :from   :search
+   :where
+   (into
+    [:or]
+    (for [model ["card"]] ; just card for now
+      (search-filters-for-model model search-ctx)))})
```

Comment:
ehmmm why do we need that? Why not unified query for everyone, without looking at the model at this point?

Reply:
I'm not sure I understand what you mean by "without looking at the model", because this function doesn't know any details about the model queries. I did hard-code that "card" is a model for demonstration purposes. 

But if you mean "a single query that every search DB will compile to execute", then yes, this had a problem, because the function knows about the `search` table which is meant to be backend-specific knowledge.

Now I've moved that logic into the backend-specific `execute-query` function:

So there is now a `general-search-query`, which is general across backends
https://github.com/metabase/metabase/blob/66e45990acb10ea42270ba65d4cc74a40141aa26/src/metabase/search/impl.clj#L377-L383

And `execute-query`:
https://github.com/metabase/metabase/blob/66e45990acb10ea42270ba65d4cc74a40141aa26/src/metabase/search/impl.clj#L464-L473

You call them together to make a search query, like this:
https://github.com/metabase/metabase/blob/66e45990acb10ea42270ba65d4cc74a40141aa26/src/metabase/search/impl.clj#L482-L488

Reply:
> If you mean "a single query that every search DB will compile to execute"

I mean function `search-fields-for-model` should not exist. So the idea is that backend (function `execute-search`) gets a search (which is search string + filters), compiles and executes it. Obviously the compilation result is backend-specific, no probs. But we don't need `general-search-query` or a loop across models with some fields.

Why are fields specific to model? What's the purpose?

Reply:
As discussed over huddle, we don't need `search-filters-for-model`. The current model-specific filter logic will be contained under `searchable-data-query-for-model`. The benefit from doing so is the query can be dead-simple and fast. [83ed956](https://github.com/metabase/metabase/pull/43986/commits/83ed956afac71f30ec1f04146e3446ddc784345a)